Jenkinsfile

```diff
@@ -0,0 +1,183 @@
+      ]
+
+      branch = checkoutDetails.GIT_BRANCH == 'origin/master' ? 'master' : checkoutDetails.GIT_BRANCH
+      commitId = checkoutDetails.GIT_COMMIT
+      commitDate = OsTools.runSafe(this, "git show -s --format=%cd --date=format:%Y%m%d-%H%M%S ${commitId}")
+
+      OsTools.runSafe(this, 'git config --global user.email sonatype-ci@sonatype.com')
+      OsTools.runSafe(this, 'git config --global user.name Sonatype CI')
+
+      version = readVersion()
+
+      def apiToken
+      withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: credentialsId,
+                        usernameVariable: 'GITHUB_API_USERNAME', passwordVariable: 'GITHUB_API_PASSWORD']]) {
+        apiToken = env.GITHUB_API_PASSWORD
+      }
+      gitHub = new GitHub(this, "${organization}/${gitHubRepository}", apiToken)
+
+      if (params.nexus_repository_manager_version) {
+        stage('Update Repository Manager Version') {
+          OsTools.runSafe(this, "git checkout ${branch}")
+          dockerImages.each { updateRepositoryManagerVersion(it.dockerFilePath) }
+          version = params.nexus_repository_manager_version
+        }
+      }
+    }
+    stage('Build') {
+      gitHub.statusUpdate commitId, 'pending', 'build', 'Build is running'
+      dockerImages.each {
+        def hash = OsTools.runSafe(this, "docker build --quiet --no-cache --tag ${it.imageTag} -f ${it.dockerFilePath} .")
+        it.imageTag = hash.split(':')[1]
+
+        if (currentBuild.result == 'FAILURE') {
+            gitHub.statusUpdate commitId, 'failure', 'build', 'Build failed'
+            return
+        } else {
+            gitHub.statusUpdate commitId, 'success', 'build', 'Build succeeded'
+        }
+      }
+    }
+    if (params.nexus_repository_manager_version) {
+      stage('Commit Repository Manager Version Update') {
+        withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'integrations-github-api',
+                        usernameVariable: 'GITHUB_API_USERNAME', passwordVariable: 'GITHUB_API_PASSWORD']]) {
+          def commitMessage = params.nexus_repository_manager_version ? "Update Repository Manager to ${params.nexus_repository_manager_version}." : ""
+          OsTools.runSafe(this, """
+            git add .
+            git commit -m '${commitMessage}'
+            git push https://${env.GITHUB_API_USERNAME}:${env.GITHUB_API_PASSWORD}@github.com/${organization}/${gitHubRepository}.git ${branch}
+          """)
+        }
+      }
+    }
+    stage('Archive') {
+      dir('build/target') {
+        dockerImages.each {
+            OsTools.runSafe(this, "docker save ${it.imageTag} | gzip > ${archiveName}-${it.flavor}.tar.gz")
+        }
+        archiveArtifacts artifacts: "${archiveName}-*.tar.gz", onlyIfSuccessful: true
+      }
+    }
+    if (branch != 'master') {
+      return
+    }
+    input 'Push image and tags?'
+    stage('Push image and tags') {
+      def dockerhubApiToken
```

Comment:
This can be declared where it's set, doesn't need to reside outside the withCredentials closure

Reply:
In: 52d8379